Source/JavaScriptCore/runtime/MicrotaskQueue.h

```diff
@@ -33,37 +33,65 @@
 
 namespace JSC {
 
+class JSGlobalObject;
+class MarkedMicrotaskDeque;
+class MicrotaskDispatcher;
 class MicrotaskQueue;
 class VM;
 
 class QueuedTask {
     WTF_MAKE_TZONE_ALLOCATED(QueuedTask);
     friend class MicrotaskQueue;
+    friend class MarkedMicrotaskDeque;
 public:
     static constexpr unsigned maxArguments = 4;
 
-    QueuedTask(MicrotaskIdentifier identifier, JSValue job, JSValue argument0, JSValue argument1, JSValue argument2, JSValue argument3)
-        : m_identifier(identifier)
+    enum class Status : uint8_t {
```

Comment:
Hm... how about we call this `Result`? `Status` sounds like something that describes the current status of a task. Here, this enum is more like a result of trying to execute a task.